### ISSUE TITLE
implement thread-local storage when not using LLVM or LLD

### ISSUE THREAD

**Issue by andrewrk** [ziglang/zig#17794]
```zig
const std = @import("std");

test "thread local variables" {
    var value1: i32 = 0;
    var value2: i32 = 0;

    const thread1 = try std.Thread.spawn(.{}, run, .{&value1});
    const thread2 = try std.Thread.spawn(.{}, run, .{&value2});

    thread1.join();
    thread2.join();

    try std.testing.expectEqual(@as(i32, 1235), value1);
    try std.testing.expectEqual(@as(i32, 1235), value2);
    try std.testing.expectEqual(@as(i32, 1234), x);
}

threadlocal var x: i32 = 1234;

fn run(value: *i32) void {
    x += 1;
    value.* = x;
}
```

```
$ stage3/bin/zig test test.zig  -fllvm -flld
All 1 tests passed.

$ stage3/bin/zig test test.zig  -fno-llvm -fno-lld -fno-single-threaded
Test [1/1] test.thread local variables... expected 1235, found 1236
Test [1/1] test.thread local variables... FAIL (TestExpectedEqual)
0 passed; 0 skipped; 1 failed.
error: the following test command failed with exit code 1:
/home/andy/Downloads/zig/zig-cache/o/a01721253d2ad18ca1ea314f4ec3169e/test
```

**Comment by andrewrk** [comment]
https://github.com/ziglang/zig/blob/98dc28bbe223cb7183aabe7ed7a847c67c1a4df9/src/Compilation.zig#L1124-L1129

**Comment by kubkon** [comment]
Implemented by #17978.

```
$ stage3/bin/zig test test.zig -fno-llvm -fno-lld -fno-single-threaded
All 1 tests passed.
$ stage3/bin/zig test test.zig -fno-llvm -fno-lld -fno-single-threaded -fPIC
All 1 tests passed.
```
